spinnaker/spinnaker_system/google_http_lb_test.py

```diff
@@ -0,0 +1,1256 @@
+    'HEALTH_CHECKS': 1,
+    'IN_USE_ADDRESSES': 2,
+    'SSL_CERTIFICATES': 2,
+    'TARGET_HTTP_PROXIES': 1,
+    'TARGET_HTTPS_PROXIES': 1,
+    'FORWARDING_RULES': 2
+  }
+
+  MINIMUM_REGION_QUOTA = {
+      'CPUS': 2,
+      'IN_USE_ADDRESSES': 2,
+      'INSTANCE_GROUP_MANAGERS': 1,
+      'INSTANCES': 2,
+  }
+
+  @classmethod
+  def new_agent(cls, bindings):
+    '''Implements citest.service_testing.AgentTestScenario.new_agent.'''
+    return gate.new_agent(bindings)
+
+
+  @classmethod
+  def initArgumentParser(cls, parser, defaults=None):
+    '''Initialize command line argument parser.
+
+    Args:
+      parser: argparse.ArgumentParser
+    '''
+    super(GoogleHttpLoadBalancerTestScenario, cls).initArgumentParser(
+      parser, defaults=defaults)
+
+
+  def __init__(self, bindings, agent=None):
+    '''Constructor.
+
+    Args:
+      bindings: [dict] The data bindings to use to configure the scenario.
+      agent: [GateAgent] The agent for invoking the test operations on Gate.
+    '''
+    super(GoogleHttpLoadBalancerTestScenario, self).__init__(bindings, agent)
+
+    bindings = self.bindings
+
+    self.__lb_detail = 'lb'
+    self.TEST_APP = bindings['TEST_APP']
+    self.__lb_name = '{app}-{stack}-{detail}'.format(
+        app=bindings['TEST_APP'], stack=bindings['TEST_STACK'],
+        detail=self.__lb_detail)
+    self.__default_hc = {
+      'name': 'basic-http-check',
+      'requestPath': '/',
+      'port': 80,
+      'checkIntervalSec': 1,
+      'timeoutSec': 1,
+      'healthyThreshold': 1,
+      'unhealthyThreshold': 1
+    }
+    self.__default_upsert = [
+      {
+        'cloudProvider': 'gce',
+        'provider': 'gce',
+        'stack': bindings['TEST_STACK'],
+        'credentials': bindings['GCE_CREDENTIALS'],
+        'region': bindings['TEST_GCE_REGION'],
+        'loadBalancerType': 'HTTP',
+        'loadBalancerName': self.__lb_name,
+        'portRange': '80',
+        'defaultService': {
+          'name': 'default-backend-service',
+          'backends': [],
+          'healthCheck': self.__default_hc,
+        },
+        'certificate': '',
+        'hostRules': [
+          {
+            'hostPatterns': ['host1.com', 'host2.com'],
+            'pathMatcher': {
+              'pathRules': [
+                {
+                  'paths': ['/path', '/path2/more'],
+                  'backendService': {
+                    'name': 'backend-service',
+                    'backends': [],
+                    'healthCheck': self.__default_hc,
+                  }
+                }
+              ],
+              'defaultService': {
+                'name': 'pm-backend-service',
+                'backends': [],
+                'healthCheck': self.__default_hc,
+              }
+            }
+          }
+        ],
+        'type': 'upsertLoadBalancer',
+        'availabilityZones': {bindings['TEST_GCE_REGION']: []},
+        'user': '[anonymous]'
+      }
+    ]
+    self.__default_delete = [
```

Comment:
make this just a single instance and wrap it in a list where the payload wants a list.